### PR TITLE
Add table azure_lb_backend_address_pool. Closes #183

### LINKED ISSUE #183: Add table azure_lb_backend_address_pool
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

### PR DESCRIPTION
**Pull request by c0d3r-arnab**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
arnab@turbotindias-MacBook-Pro azure-test % ./tint.js azure_lb_backend_address_pool
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/azure_lb_backend_address_pool []

PRETEST: tests/azure_lb_backend_address_pool

TEST: tests/azure_lb_backend_address_pool
Running terraform

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # azurerm_lb.named_test_resource will be created
  + resource "azurerm_lb" "named_test_resource" {
      + id                   = (known after apply)
      + location             = "westus"
      + name                 = "turbottest96345"
      + private_ip_address   = (known after apply)
      + private_ip_addresses = (known after apply)
      + resource_group_name  = "turbottest96345"
      + sku                  = "Basic"
      + tags                 = (known after apply)

      + frontend_ip_configuration {
          + inbound_nat_rules             = (known after apply)
          + load_balancer_rules           = (known after apply)
          + name                          = "turbottest96345"
          + outbound_rules                = (known after apply)
          + private_ip_address            = (known after apply)
          + private_ip_address_allocation = (known after apply)
          + public_ip_address_id          = (known after apply)
          + public_ip_prefix_id           = (known after apply)
          + subnet_id                     = (known after apply)
        }
    }

  # azurerm_lb_backend_address_pool.named_test_resource will be created
  + resource "azurerm_lb_backend_address_pool" "named_test_resource" {
      + backend_ip_configurations = (known after apply)
      + id                        = (known after apply)
      + load_balancing_rules      = (known after apply)
      + loadbalancer_id           = (known after apply)
      + name                      = "turbottest96345"
      + resource_group_name       = "turbottest96345"
    }

  # azurerm_public_ip.named_test_resource will be created
  + resource "azurerm_public_ip" "named_test_resource" {
      + allocation_method            = "Static"
      + fqdn                         = (known after apply)
      + id                           = (known after apply)
      + idle_timeout_in_minutes      = 4
      + ip_address                   = (known after apply)
      + ip_version                   = "IPv4"
      + location                     = "westus"
      + name                         = "turbottest96345"
      + public_ip_address_allocation = (known after apply)
      + resource_group_name          = "turbottest96345"
      + sku                          = "Basic"
      + tags                         = (known after apply)
    }

  # azurerm_resource_group.named_test_resource will be created
  + resource "azurerm_resource_group" "named_test_resource" {
      + id       = (known after apply)
      + location = "westus"
      + name     = "turbottest96345"
      + tags     = (known after apply)
    }

Plan: 4 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + resource_aka       = (known after apply)
  + resource_aka_lower = (known after apply)
  + resource_id        = (known after apply)
  + resource_name      = "turbottest96345"
  + subscription_id    = "d46d7416-f95f-4771-bbb5-529d4c76659c"
azurerm_resource_group.named_test_resource: Creating...
azurerm_resource_group.named_test_resource: Creation complete after 3s [id=/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest96345]
azurerm_public_ip.named_test_resource: Creating...
azurerm_public_ip.named_test_resource: Still creating... [10s elapsed]
azurerm_public_ip.named_test_resource: Creation complete after 12s [id=/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest96345/providers/Microsoft.Network/publicIPAddresses/turbottest96345]
azurerm_lb.named_test_resource: Creating...
azurerm_lb.named_test_resource: Creation complete after 10s [id=/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest96345/providers/Microsoft.Network/loadBalancers/turbottest96345]
azurerm_lb_backend_address_pool.named_test_resource: Creating...
azurerm_lb_backend_address_pool.named_test_resource: Creation complete after 6s [id=/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest96345/providers/Microsoft.Network/loadBalancers/turbottest96345/backendAddressPools/turbottest96345]

Warning: Version constraints inside provider configuration blocks are deprecated

  on variables.tf line 21, in provider "azurerm":
  21:   version         = "=1.36.0"

Terraform 0.13 and earlier allowed provider version constraints inside the
provider configuration block, but that is now deprecated and will be removed
in a future version of Terraform. To silence this warning, move the provider
version constraint into the required_providers block.

Apply complete! Resources: 4 added, 0 changed, 0 destroyed.

Outputs:

resource_aka = "azure:///subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest96345/providers/Microsoft.Network/loadBalancers/turbottest96345/backendAddressPools/turbottest96345"
resource_aka_lower = "azure:///subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourcegroups/turbottest96345/providers/microsoft.network/loadbalancers/turbottest96345/backendaddresspools/turbottest96345"
resource_id = "/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest96345/providers/Microsoft.Network/loadBalancers/turbottest96345/backendAddressPools/turbottest96345"
resource_name = "turbottest96345"
subscription_id = "d46d7416-f95f-4771-bbb5-529d4c76659c"

Running SQL query: test-get-query.sql
[
  {
    "id": "/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest96345/providers/Microsoft.Network/loadBalancers/turbottest96345/backendAddressPools/turbottest96345",
    "name": "turbottest96345",
    "resource_group": "turbottest96345",
    "subscription_id": "d46d7416-f95f-4771-bbb5-529d4c76659c",
    "type": "Microsoft.Network/loadBalancers/backendAddressPools"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "id": "/subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest96345/providers/Microsoft.Network/loadBalancers/turbottest96345/backendAddressPools/turbottest96345",
    "name": "turbottest96345"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "azure:///subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbottest96345/providers/Microsoft.Network/loadBalancers/turbottest96345/backendAddressPools/turbottest96345",
      "azure:///subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourcegroups/turbottest96345/providers/microsoft.network/loadbalancers/turbottest96345/backendaddresspools/turbottest96345"
    ],
    "name": "turbottest96345",
    "title": "turbottest96345"
  }
]
✔ PASSED

POSTTEST: tests/azure_lb_backend_address_pool

TEARDOWN: tests/azure_lb_backend_address_pool

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  id,
  name,
  load_balancer_name,
  provisioning_state,
  type
from
  azure_lb_backend_address_pool;
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------
| id                                                                                                                                                                                    | name              
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------
| /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/hipaa-hitrust/providers/Microsoft.Network/loadBalancers/test_load_balancer/backendAddressPools/test_backend_pool_1 | test_backend_pool_
| /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/hipaa-hitrust/providers/Microsoft.Network/loadBalancers/test_load_balancer/backendAddressPools/test_backend_pool_2 | test_backend_pool_
| /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbot_rg/providers/Microsoft.Network/loadBalancers/my_lb/backendAddressPools/my_backend_pool                      | my_backend_pool   
| /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/turbot_rg/providers/Microsoft.Network/loadBalancers/teret/backendAddressPools/test                                 | test              
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------
> select
  id,
  name,
  type,
  provisioning_state
from
  azure_lb_backend_address_pool
where
  provisioning_state = 'Failed';
+----+------+------+--------------------+
| id | name | type | provisioning_state |
+----+------+------+--------------------+
+----+------+------+--------------------+
> select
  id,
  name,
  type,
  provisioning_state
from
  azure_lb_backend_address_pool
where
  load_balancer_name = 'test_load_balancer' and name = 'test_backend_pool_1';
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------
| id                                                                                                                                                                                    | name              
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------
| /subscriptions/d46d7416-f95f-4771-bbb5-529d4c76659c/resourceGroups/hipaa-hitrust/providers/Microsoft.Network/loadBalancers/test_load_balancer/backendAddressPools/test_backend_pool_1 | test_backend_pool_
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------------
>
```
</details>

### CODE REVIEW

File: azure-test/tests/azure_lb_backend_address_pool/test-get-query.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+select name, id, type, resource_group, subscription_id
+from azure.azure_lb_backend_address_pool
+where name = '{{ resourceName }}' and resource_group = '{{ resourceName }}';
```
Comment (bigdatasourav) [inline azure-test/tests/azure_lb_backend_address_pool/test-get-query.sql:3]: Get call require 3 columns as per the table , update and rerun the test

File: azure/table_azure_lb_backend_address_pool.go [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "outbound_rule_id",
+				Description: "A reference to an outbound rule that uses this backend address pool.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("BackendAddressPoolPropertiesFormat.OutboundRule.ID"),
+			},
+			{
+				Name:        "backend_addresses",
```
Comment (bigdatasourav) [inline azure/table_azure_lb_backend_address_pool.go:70]: Column should be as per the api response

File: azure/table_azure_lb_backend_address_pool.go [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+				Transform:   transform.FromField("BackendAddressPoolPropertiesFormat.LoadBalancingRules"),
+			},
+			{
+				Name:        "outbound_rules",
+				Description: "An array of references to outbound rules that use this backend address pool.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("BackendAddressPoolPropertiesFormat.OutboundRules"),
+			},
+			{
+				Name:        "tunnel_interfaces",
```
Comment (bigdatasourav) [inline azure/table_azure_lb_backend_address_pool.go:94]: column should be as per the api response

File: azure/table_azure_lb_backend_address_pool.go [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("ID").Transform(idToAkas),
+			},
+
+			// Azure standard columns
+			{
+				Name:        "region",
+				Description: ColumnDescriptionRegion,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("BackendAddressPoolPropertiesFormat.Location").Transform(toLower),
```
Comment (bigdatasourav) [inline azure/table_azure_lb_backend_address_pool.go:119]: does location coming under properties?

File: docs/tables/azure_lb_backend_address_pool.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  name,
+  type,
+  provisioning_state
+from
+  azure_lb_backend_address_pool
+where
+  provisioning_state = 'Failed';
+```
+
+### Get load balancer backend address pool by load balancer name and backend address pool name
```
Comment (bigdatasourav) [inline docs/tables/azure_lb_backend_address_pool.md:34]: remove this example

File: docs/tables/azure_lb_backend_address_pool.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
+# Table: azure_lb_backend_address_pool
+
+The backend pool is a critical component of the load balancer. The backend pool defines the group of resources that will serve traffic for a given load-balancing rule.
```
Comment (bigdatasourav) [inline docs/tables/azure_lb_backend_address_pool.md:3]:
```suggestion
An Azure load balancer's backend address pool consists of IP addresses associated with the virtual machine NICs. This pool is used to distribute traffic to the virtual machines behind the load balancer. 
```

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Pull request merged**